### PR TITLE
refactor: broke down identities.register and identities.topUp into a number of exportable methods

### PR DESCRIPTION
**Pull request by antouhou**
## Issue being fixed or feature implemented

Break down identity register and identity top-up methods into reusable functions

## What was done?

- added `createAssetLockProof` function;
- added `createIdentityCreateTransition` function;
- added `createIdentityTopUp` function;

## How Has This Been Tested?

Add a new test for the top-up method; Rerun old register tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by jawid-h — APPROVED** [review]
Sweet :candy:

**Review by Alex-Werner — DISMISSED** [review]
Cool !

File: src/SDK/Client/Platform/methods/identities/internal/createAssetLockProof.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    // Create poof that the transaction won't be double spend
+    if (passFakeAssetLockProofForTests) {
+        instantLock = createFakeInstantLock(assetLockTransaction.hash);
+    } else {
+        instantLock = await account.waitForInstantLock(assetLockTransaction.hash);
+    }
+
+    // @ts-ignore
+    return await dpp.identity.createInstantAssetLockProof(instantLock);
+}
```
Comment (shumkov) [inline src/SDK/Client/Platform/methods/identities/internal/createAssetLockProof.ts:26]: new line

**Comment by thephez** [comment]
Filename typo: 
`<...>/identities/internal/createIdnetityTopUpTransition.ts` should be 
`<...>/identities/internal/createIdentityTopUpTransition.ts`

**Pull request merged**